Device/Device.cs

```diff
@@ -65,6 +66,11 @@ public MqttApplicationMessage ConstructMessage(string topic, string payload, boo
             return message;
         }
 
+        public MqttApplicationMessage ConstructWillMessage(string willPayload, bool retainFlag = true)
+        {
+            var willTopic = $"devices/{deviceId}/messages/events/$.ct=application%2Fjson&$.ce=utf-8";

```

Comment:
The `willTopic` is exactly the same as `topicD2C` in the SendDeviceToCloudMessageAsync() method. Should we extract it as a private class member and reuse?

Reply:
Yes, that makes sense.